internal/broker/broker_test.go

```diff
@@ -827,6 +831,147 @@ func TestConcurrentIsAuthenticated(t *testing.T) {
 	}
 }
 
+func TestIsAuthenticatedAllowedUsersConfig(t *testing.T) {
+	t.Parallel()
+	tests := map[string]struct {
+		allowedUsers          map[string]struct{}
+		owner                 string
+		ownerAllowed          bool
+		allUsersAllowed       bool
+		firstUserBecomesOwner bool
+
+		allowedUsernames   []string
+		unallowedUsernames []string
+	}{
+		"All_users_are_allowed": {
+			allUsersAllowed:  true,
+			allowedUsernames: []string{"u1", "u2"},
+		},
+		"Only_owner_allowed": {
+			ownerAllowed:       true,
+			owner:              "machine_owner",
+			allowedUsernames:   []string{"machine_owner"},
+			unallowedUsernames: []string{"u1", "u2"},
+		},
+		"No_users_allowed": {
+			ownerAllowed:       true,
+			owner:              "",
+			unallowedUsernames: []string{"u1", "u2", "machine_owner", "u3"},
+		},
+		"Only_unset_owner_allowed": {
```

Comment:
Maybe a bit easier to understand

```suggestion
		"Only_first_user_allowed": {
```